packages/vite-plugin/src/index.ts

```diff
@@ -60,50 +55,48 @@ export function vanillaExtractPlugin({ identifiers }: Options = {}): Plugin {
       }
 
       virtualExt = `.vanilla.${config.command === 'serve' ? 'js' : 'css'}`;
-
-      packageInfo = getPackageInfo(config.root);
     },
     resolveId(id) {
-      if (id.indexOf(virtualPrefix) === 0) {
-        return id;
+      if (!id.endsWith(virtualExt)) {
+        return;
+      }
+
+      const normalizedId = id.startsWith('/') ? id.slice(1) : id;
+
+      if (cssMap.has(normalizedId)) {
+        return normalizePath(path.join(config.root, normalizedId));
       }
     },
     load(id) {
-      if (id.indexOf(virtualPrefix) === 0) {
-        const fileScopeId = id.slice(
-          virtualPrefix.length,
-          id.indexOf(virtualExt),
-        );
-
-        if (!cssMap.has(fileScopeId)) {
-          throw new Error(`Unable to locate ${fileScopeId} in the CSS map.`);
-        }
-
-        const css = cssMap.get(fileScopeId)!;
+      if (!id.endsWith(virtualExt)) {
+        return;
+      }
 
-        if (!server) {
-          return css;
-        }
+      const cssFileId = id.slice(config.root.length + 1);
+      const css = cssMap.get(cssFileId);
 
-        const fileScope = parseFileScope(fileScopeId);
+      if (!css) {
```

Comment:
I was trying this code out locally and ran into a case where an empty CSS file would throw an error trying to load the `.css.ts.vanilla.css` file. This would allow an empty string to get through :)
```suggestion
      if (typeof css !== 'undefined') {
```